src/testRunner/unittests/tsserver/projectReferencesSourcemap.ts

```diff
@@ -549,4 +550,51 @@ fn5();
             });
         }, /*options*/ undefined);
     });
+
+    verifyForAllSessionTypes(type => {
+        it("goto Definition in usage of a property with mapped type origin", () => {
+            const dependencyTs: File = {
+                path: `${dependencyLocation}/api.ts`,
+                content: dedent`
+                    type ValidateShape<T> = {
+                        [K in keyof T]: T[K];
+                    };
+
+                    function getApi<T>(arg: ValidateShape<T>) {
+                        function createCaller<T>(arg: T): () => {
+                            [K in keyof T]: () => T[K];
+                        } {
+                            return null as any;
+                        }
+                        return {
+                            createCaller: createCaller(arg),
+                        };
+                    }
+
+                    const obj = getApi({
+                        foo: 1,
+                        bar: "",
+                    });
+
+                    export const createCaller = obj.createCaller;
+                `,
+            };
+            const mainTs: File = {
+                path: `${mainLocation}/main.ts`,
+                content: dedent`
+                    import { createCaller } from "../decls/api";
+                    const caller = createCaller();
+                    caller.foo;
+                `,
+            };
+            const files = [dependencyTs, dependencyConfig, mainTs, mainConfig];
+            const session = createSession(type, files);
+            openFilesForSession([mainTs], session);
+            session.executeCommandSeq<ts.server.protocol.DefinitionAndBoundSpanRequest>({
+                command: ts.server.protocol.CommandTypes.DefinitionAndBoundSpan,
+                arguments: { file: mainTs.path, line: 3, offset: "caller.foo".length - 2 },

```

Comment:
Pls use `protocolFileLocationFromSubstring` to get line and offset

Reply:
done